src/main/java/baseball/utils/UserInputValidator.java

```diff
@@ -0,0 +1,73 @@
+package baseball.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserInputValidator {
+
+    private static final String NOT_DIFFERENT_NUMBERS_MSG = "서로 다른 숫자를 입력해주세요.";
+    private static final String WRONG_NUMBER_RANGE_MSG = "1 ~ 9까지의 숫자를 입력해주세요.";
+    private static final String WRONG_NUMBER_LENGTH_MSG = "3 자리의 숫자를 입력해주세요.";
+    private static final String WRONG_RESTART_NUMBER_LENGTH_MSG = "1 자리의 숫자를 입력해주세요.";
+    private static final String WRONG_RESTART_NUMBER_RANGE_MSG = "1 또는 2를 입력해주세요.";
+    private static final int NUMBER_LENGTH = 3;
+    private static final int RESTART_NUMBER_LENGTH = 1;
+    private static final char MIN_NUMBER = '1';
+    private static final char MAX_NUMBER = '9';
+    private String userInputNumbers = "";
+    private String userInputRestartNumber = "";
+    private static final String RESTART_NUMBER = "1";
+    private static final String DO_NOT_RESTART_NUMBER = "2";
+
+    public void validateInputNumbers(String userInput) {
+        userInputNumbers = userInput;
+
+        checkDifferentNumbers();
+        checkNumberRange();
+        checkNumberLength();
+    }
+
+    private void checkNumberLength() {
+        if (userInputNumbers.length() != NUMBER_LENGTH) {
+            throw new IllegalArgumentException(WRONG_NUMBER_LENGTH_MSG);
+        }
+    }
+
+    private void checkNumberRange() {
+        for (char number : userInputNumbers.toCharArray()) {
+            if (number < MIN_NUMBER || number > MAX_NUMBER) {
+                throw new IllegalArgumentException(WRONG_NUMBER_RANGE_MSG);
+            }
+        }
+    }
+
+    private void checkDifferentNumbers() {
+        List<Character> numbers = new ArrayList<>();
+        for (char number : userInputNumbers.toCharArray()) {
+            if (!numbers.contains(number)) {
+                numbers.add(number);
+            } else {
+                throw new IllegalArgumentException(NOT_DIFFERENT_NUMBERS_MSG);
+            }
+        }
```

Comment:
```suggestion
        for (char number : userInputNumbers.toCharArray()) {
            if (numbers.contains(number)) {
                throw new IllegalArgumentException(NOT_DIFFERENT_NUMBERS_MSG);
            }
            
             numbers.add(number);
        }
```
이거 if문을 뒤집으면 `early-return` 식으로 작성이 되서 더 코드가 깔끔해질 것 같습니다.

Reply:
다음부턴 early-return이랑 else문을 없애는 쪽으로도 생각해봐야겠군요